presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/CombineApproxPercentileFunctions.java

```diff
@@ -0,0 +1,315 @@
+        // approx_percentile(x, percentage) -> arguments.size() == 2
+        // approx_percentile(x, percentage, accuracy) -> arguments.size() == 3 && arguments.get(1) is Double
+        // approx_percentile(x, w, percentage) -> arguments.size() == 3 && arguments.get(1) is BigInt
+        // approx_percentile(x, w, percentage, accuracy) -> arguments.size() == 4
+        checkState(functionHandle instanceof BuiltInFunctionHandle);
+        List<TypeSignature> argumentTypes = ((BuiltInFunctionHandle) functionHandle).getSignature().getArgumentTypes();
+        if (argumentTypes.size() == 2 || (argumentTypes.size() == 3 && argumentTypes.get(1).getBase().equals(StandardTypes.DOUBLE))) {
+            return 1;
+        }
+        checkState(argumentTypes.size() == 4 || (argumentTypes.size() == 3 && argumentTypes.get(1).getBase().equals(StandardTypes.BIGINT)));
+        return 2;
+    }
+
+    private static boolean aggregationCanMerge(AggregationNode.Aggregation aggregation1, AggregationNode.Aggregation aggregation2)
+    {
+        if (!aggregation1.getMask().equals(aggregation2.getMask())
+                || !aggregation1.getOrderBy().equals(aggregation2.getOrderBy())
+                || !aggregation1.getFilter().equals(aggregation2.getFilter())
+                || aggregation1.isDistinct() != aggregation2.isDistinct()) {
+            return false;
+        }
+        // Check call expression, the only difference should be percentile argument
+        CallExpression expression1 = aggregation1.getCall();
+        CallExpression expression2 = aggregation2.getCall();
+        int percentilePosition = getPercentilePosition(expression1.getFunctionHandle());
+        if (!expression1.getFunctionHandle().equals(expression2.getFunctionHandle()) || expression1.getArguments().size() != expression2.getArguments().size()) {
+            return false;
+        }
+        List<RowExpression> arguments1 = expression1.getArguments();
+        List<RowExpression> arguments2 = expression2.getArguments();
+        for (int i = 0; i < arguments1.size(); ++i) {
+            if (i != percentilePosition && !arguments1.get(i).equals(arguments2.get(i))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private static List<RowExpression> changePercentileArgument(List<RowExpression> arguments, RowExpression percentileArgument, int percentilePosition)
+    {
+        ImmutableList.Builder<RowExpression> newAggCallArguments = new ImmutableList.Builder<>();
+        for (int i = 0; i < arguments.size(); ++i) {
+            if (i == percentilePosition) {
+                newAggCallArguments.add(percentileArgument);
+            }
+            else {
+                newAggCallArguments.add(arguments.get(i));
+            }
+        }
+        return newAggCallArguments.build();
+    }
+
+    private static List<List<AggregationNode.Aggregation>> createMergeableAggregations(List<AggregationNode.Aggregation> candidateAggregations)
```

Comment:
Just add brief description that the output is a list of list of aggs that will be merged.